lib/sinatra/activerecord.rb

```diff
@@ -16,7 +16,19 @@ def database
 
   module ActiveRecordExtension
     def self.registered(app)
-      if ENV['DATABASE_URL']
+      if ENV['DATABASE_URL'] && File.exist?("#{Dir.pwd}/config/database.yml")
+        path = "#{Dir.pwd}/config/database.yml"
+        url = ENV['DATABASE_URL']
+        file_path = File.join(root, path) if Pathname(path).relative? and root
+        file_spec = YAML.load(ERB.new(File.read(path)).result) || {}
+
+        url_spec = ActiveRecord::ConnectionAdapters::ConnectionSpecification::ConnectionUrlResolver.new(url).to_hash
+
+        # url_spec will override the same key, if exist
+        final_spec = file_spec.keys.map{ |env| [env, file_spec[env].merge(url_spec)] }.to_h
```

Comment:
what if there are nested keys? this will only override the top-level keys, right?

Reply:
Make sense, hmm I think the top level keys from the database.yml will usually refer to environment usually (development, staging, production etc), whereas database URL dont have environment specified, should be fine I think

Reply:
[Using multiple databases](https://guides.rubyonrails.org/active_record_multiple_databases.html) does use nested keys (and there could certainly be other use cases I'm not aware of). That link is for Rails, but as far as I can tell the feature isn't Rails-specific. The top level keys would still be environment names, but this code would merge `url_spec` into `production` rather than `production['primary']`.

I'm not 100% sure sinatra-activerecord supports the multiple database configuration currently, though. If not, then it's a non-issue for this PR.

Reply:
I have just tested multiple database configuration, and the test fails, I think this gem doesn't support multiple database previously, plus multiple database is a new feature for ActiveRecord 6, I feel this PR shouldn't affect current users.